nibabel/cifti2/cifti2_axes.py

```diff
@@ -0,0 +1,1330 @@
+    def volume_shape(self, ):
+        """
+        Shape of the volumetric image in which the greyordinate voxels were defined
+        """
+        return self._volume_shape
+
+    @volume_shape.setter
+    def volume_shape(self, value):
+        if value is not None:
+            value = tuple(value)
+            if len(value) != 3:
+                raise ValueError("Volume shape should be a tuple of length 3")
+            if not all(isinstance(v, integer_types) for v in value):
+                raise ValueError("All elements of the volume shape should be integers")
+        self._volume_shape = value
+
+    _name = None
+
+    @property
+    def name(self, ):
+        """The brain structure to which the voxel/vertices of belong
+        """
+        return self._name
+
+    @name.setter
+    def name(self, values):
+        self._name = np.array([self.to_cifti_brain_structure_name(name) for name in values])
+
+    def __len__(self):
+        return self.name.size
+
+    def __eq__(self, other):
+        if not isinstance(other, BrainModel) or len(self) != len(other):
+            return False
+        if xor(self.affine is None, other.affine is None):
+            return False
+        return (
+                ((self.affine is None and other.affine is None) or
+                 (abs(self.affine - other.affine).max() < 1e-8 and
+                  self.volume_shape == other.volume_shape)) and
+                (self.nvertices == other.nvertices) and
+                (self.name == other.name).all() and
+                (self.voxel[~self.is_surface] == other.voxel[~other.is_surface]).all() and
+                (self.vertex[~self.is_surface] == other.vertex[~other.is_surface]).all()
+        )
+
+    def __add__(self, other):
+        """
+        Concatenates two BrainModels
+
+        Parameters
+        ----------
+        other : BrainModel
+            brain model to be appended to the current one
+
+        Returns
+        -------
+        BrainModel
+        """
+        if isinstance(other, BrainModel):
+            if self.affine is None:
+                affine, shape = other.affine, other.volume_shape
+            else:
+                affine, shape = self.affine, self.volume_shape
+                if other.affine is not None and (
+                        (other.affine != affine).all() or
+                        other.volume_shape != shape
+                ):
+                    raise ValueError("Trying to concatenate two BrainModels defined " +
+                                     "in a different brain volume")
+            nvertices = dict(self.nvertices)
+            for name, value in other.nvertices.items():
+                if name in nvertices.keys() and nvertices[name] != value:
+                    raise ValueError("Trying to concatenate two BrainModels with inconsistent " +
+                                     "number of vertices for %s" % name)
+                nvertices[name] = value
+            return type(self)(
```

Comment:
I think we generally use the `__class__` approach for methods like these:

```suggestion
            return self.__class__(
```